### DISCUSSION TITLE
Cannot use Vercel Serverless Function & Nuxt Server APIs in the same project

### DISCUSSION THREAD

**Discussion by Tiarhai** [vercel/community#4332] [Help] [answered]
### Summary

Hi, 

I believe I must be missing something basic, but I can't figure it out.

When I add a Vercel Serverless Function (to avail of the new 300s timeout), it causes all of my Nuxt Server APIs to 404.

### Steps to Reproduce

Starting from a standard Nuxt 3 project.

**/api/hello-world.js**
```
export default async function handler(request, response) {
  return response.status(200).json({ message: "Hello world!" });
}
```

**vercel.json**
```
{
  "functions": {
    "api/hello-world.js": {
      "maxDuration": 300
    }
  }
}
```

**Standard Nuxt server API**
**/server/api/test.js**
```
export default eventHandler(async () => {
  return "Test";
});
```

Once deployed, I see in the Deployment Summary on Vercel, that two Functions have been deployed: `__nitro` and `api/hello-world`.

Without the Vercel Serverless Function, `__nitro` is the only Function deployed, and `/api/test` will return "Test" as expected. 

However, with the Vercel Serverless Function, only the `/api/hello-world` route will work, and Nuxt's `/api/test` route will 404.

I feel as though I am missing something obvious in order to get both types of APIs to work, but I can't for the life of me find what I am doing wrong.

Thank you in advance,
Gary

**Edit:**

I turned on Directory Listing, and only the Vercel Serverless Function "hello-world" is listed in the /api/ directory. To what route has the Nuxt server API been deployed? I have no idea.

**Comment by swarnava** [comment]
Hi,

I will recommend reaching out to [Nitro community](https://github.com/unjs/nitro/issues) to check if they support [`maxDuration`](https://vercel.com/docs/functions/serverless-functions/runtimes#maxduration) parameter for Nuxt project. I have forwarded this internally as well.

In the meantime, if you send me the link of the project or any deployment URL, we can increase the timeout value to the desire value you want.

**Comment by swarnava** [comment, marked as answer]
Hi, 

You can do it using following method:

```js
import { defineNuxtConfig } from "nuxt/config";

export default defineNuxtConfig({
  nitro: {
    vercel: {
      functions: {
        maxDuration: 60,
        memory: 1024,
      },
    },
  },
});
```
